src/widgets/highcharts/ui/DailyTaskActivityChart/DailyTaskActivityChart.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {
+    memo, useMemo, useRef, useState,
+} from 'react';
+import {
+    Box, FormControl, InputLabel, MenuItem, Select, SelectChangeEvent,
+} from '@mui/material';
+import Highcharts from 'highcharts';
+import HighchartsReact from 'highcharts-react-official';
+import { useTranslation } from 'react-i18next';
+import { useAppSelector } from 'shared/lib/hooks/redux';
+import { getTasksData } from 'entities/Task';
+import { getDataForChart } from 'shared/lib/helpers';
+import { useTheme } from 'app/providers/AppThemeProvider';
+import { addColorSchemeToHighcharts } from '../../lib/addColorSchemeToHighcharts';
+import { DailyTaskActivityChartStyle as styles } from './DailyTaskActivityChart.style';
+
+interface Band {
+    color: string;
+    from: number;
+    to: number;
+}
+
+export const DailyTaskActivityChart = memo(() => {
+    const tasks = useAppSelector(getTasksData);
+    const chartComponentRef = useRef<HighchartsReact.RefObject>(null);
+    const { t } = useTranslation('translation');
+    const { theme: { palette: { mode } } } = useTheme();
+    const currentDate = new Date(Date.now());
+
+    const [year, setYear] = useState<number>(currentDate.getFullYear());
+    const [month, setMonth] = useState<number>(currentDate.getMonth());
+
+    const handleChangeMonth = (event: SelectChangeEvent) => {
+        setMonth(Number(event.target.value));
+    };
+
+    const handleChangeYear = (event: SelectChangeEvent) => {
+        setYear(Number(event.target.value));
+    };
+
+    const months = [
+        'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec',
+    ];
+
+    const {
+        dates, createdTasksCount, completedTasksCount,
+    } = useMemo(() => getDataForChart(month, year, tasks), [year, month, tasks]);
+
+    const isWeekend = (date: Date) => {
+        const day = date.getDay();
+        return day === 0 || day === 6;
+    };
```

Comment:
utils like this is better to store separately from the component :)

Reply:
Done